File: php/googleae/email-webhook/incoming-email-webhook.py
import logging, email, urllib
from google.appengine.ext import webapp 
from google.appengine.ext.webapp.mail_handlers import InboundMailHandler 
from google.appengine.ext.webapp.util import run_wsgi_app
from google.appengine.api import urlfetch
from google.appengine.api import app_identity

class MessageHandlerException(Exception):
    """The general exception object thrown by MessageHandler"""
    def __init__(self, msg):
        self.msg = msg
		
class WebHook(InboundMailHandler):
    def receive(self, mail_message):
        logging.info("Received a message from: " + mail_message.sender)
        body = list(mail_message.bodies(content_type='text/plain'))[0]
        logging.info("Body of message: " + body[1].decode())
        
        # Point to the Main Email trigger URL which will poll for all Firehalls
        # Example replace: http://soft-haus.com/svvfd/riprunner/ with the root of you installation
        url = "http://soft-haus.com/svvfd/riprunner/webhooks/email_trigger_webhook.php"
        form_fields = {
            "sender": mail_message.sender,
            "subject": mail_message.subject,
            "to": mail_message.to,
            "date": mail_message.date,
            "body": body[1].decode()
        }
        form_data = urllib.urlencode(form_fields)
        GAE_APP_ID = app_identity.get_application_id()
        GAE_ACCOUNT_NAME = app_identity.get_service_account_name()
        logging.info("AppID: " + GAE_APP_ID + " SAM: " + GAE_ACCOUNT_NAME)
        result = urlfetch.fetch(url=url,
                        payload=form_data,
                        method=urlfetch.POST,
                        headers={'Content-Type': 'application/x-www-form-urlencoded',
                                 'X-RipRunner-Auth-APPID': GAE_APP_ID,
                                 'X-RipRunner-Auth-ACCOUNTNAME': GAE_ACCOUNT_NAME})
        logging.info(result.status_code)
        logging.info(result.content)
		
application = webapp.WSGIApplication([('/_ah/mail/.+', WebHook)],
                                     debug=True)
			
def main():
    run_wsgi_app(application)
    
if __name__ == '__main__':
    main()

File: php/googleae/email-webhook/incoming-email-webhook.php
<?php
// ==============================================================
//	Copyright (C) 2016 Mark Vejvoda
//	Under GNU GPL v3.0
// ==============================================================
ini_set('display_errors', 'On');
error_reporting(E_ALL);

use google\appengine\api\app_identity\AppIdentityService;
require_once 'PlancakeEmailParser.php';

$emailParser = new PlancakeEmailParser(file_get_contents('php://input'));
$email_body = $emailParser->getBody();
sysLog(LOG_INFO,"PHP Body of message: ".$email_body);

$GAE_APP_ID = AppIdentityService::getApplicationId();
$GAE_ACCOUNT_NAME = AppIdentityService::getServiceAccountName();
sysLog(LOG_INFO,"PHP AppID: ".$GAE_APP_ID." SAM: ".$GAE_ACCOUNT_NAME);

$data = array('sender' => $emailParser->getHeader('return-path'),
              'subject' => $emailParser->getSubject(),
              'to' => $emailParser->getTo(),
              'date' => $emailParser->getHeader('date'),
              'body' => $emailParser->getPlainBody()
        );
$data = http_build_query($data);

$context =
array("http"=>
    array(
        "method" => "post",
//        "header" => "Authorization: Basic " . base64_encode($username.':'.$password) . "\r\n",
        "header" => 'Content-Type: application/x-www-form-urlencoded'.
                    "\r\n".
                    'X-RipRunner-Auth-APPID: '.$GAE_APP_ID.
                    "\r\n".
                    'X-RipRunner-Auth-ACCOUNTNAME: '.$GAE_ACCOUNT_NAME,
        "content" => $data
    ),
    "ssl" => array(
            "allow_self_signed" => true,
            "verify_peer" => false
    )        
);
$context = stream_context_create($context);

$url = "http://soft-haus.com/svvfd/riprunner/webhooks/email_trigger_webhook.php";
$result = file_get_contents($url, false, $context);

sysLog(LOG_INFO,"PHP Result of calling webhook url: ". $result);
